csrc/scheduler/reduction.cpp

```diff
@@ -382,13 +382,9 @@ std::shared_ptr<ReductionParams> innerReductionHeuristic(
   rparams->block_dim_inner_reduction = ParallelType::TIDx;
   rparams->cross_grid_inner_reduction = gridim > 1;
   rparams->multiple_reds_per_blk = bdimy > 1;
-  bool pad_bdimx = bdimx > 16 &&
+  rparams->pad_inner_reduction_to_warp = bdimx > 16 &&
       bdimx * bdimy <
```

Comment:
Isn't this condition also strange? Shouldn't we check the number of threads *after* padding and shouldn't it be `<=`?

Reply:
nice catch! We should use `padded_bdimx`.